io/io/v7/src/TFile.cxx

```diff
@@ -97,7 +97,7 @@ class TV6Storage: public ROOT::Experimental::Internal::TFileStorageInterface {
   }
 
   void WriteMemoryWithType(std::string_view name, const void* address, TClass* cl) final {
-    fOldFile->WriteObjectAny(address, cl, name.to_string().c_str());
+     fOldFile->WriteObjectAny(address, cl, name.data());
```

Comment:
You are right. Starting C++11 `std::string` `.data()` and `.c_str()` both provide null-terminated result, but the same does not apply to `std::string_view`. The example I am looking even shows that one needs to create a temporary string and then call `.data()`.